src/cogent3/app/__init__.py

```diff
@@ -52,36 +56,67 @@ def _make_types(app) -> dict:
     return _types
 
 
-def available_apps(name_filter: str | None = None) -> Table:
+# private global to hold an ExtensionManager instance
+__apps = None
+
+
+def apps(force: bool = False) -> stevedore.ExtensionManager:
```

Comment:
rename to `get_app_manager()`. Otherwise you could have code like `cogent3.apps.apps` which does not convey it's returning an extension manager.